Dominik some docs cleanups after team feedback (#610)
0xPolygonMiden/miden-base: docs/architecture/accounts.md

### Account ID

A `~63` bits long identifier for the account ID (one field element `felt`). 

The four most significant bits specify the [account type](#account-types) - regular or faucet - and the [storage mode](#account-storage-modes) - public or private.

### Account data storage

> Storage for user-defined data. It is composed of two components.

Reviewer: ```suggestion
The storage for user-defined data is composed of two components.
```

> Storage for user-defined data. It is composed of two components.

Reviewer: ```suggestion
A storage for user-defined data that is composed of two components.
```

Reviewer: I would keep the articles for this sort of documentation. For API docs, we remove them.

Author: ok, let's do that then

1. A simple sparse Merkle tree of depth `8` which is index addressable. This provides the user with `256` `word` slots.

2. Users requiring additional storage can use the second component a `MerkleStore`. It allows users to store any Merkle structures they need. The root of the Merkle structure can be stored as a leaf in a simple sparse Merkle tree. When `AccountStorage` is serialized it checks if any of the leaves in the simple sparse Merkle tree are Merkle roots of other Merkle structures. If any Merkle roots are found then the Merkle structures will be persisted in the `AccountStorage`'s `MerkleStore`.

### Nonce

> Counter which increments whenever the account state changes. 

Reviewer: ```suggestion
A counter which increments whenever the account state changes. 
```

Nonce values must be strictly monotonically increasing and increment by any value smaller than `2^32` for every account update.

### Vault

> Asset container for an account.

Reviewer: ```suggestion
An asset container for an account.
```

> Asset container for an account.

Reviewer: ```suggestion
An asset container for an account.
```

## Account types

> There are two basic types of accounts in Miden. Regular accounts and faucets. Only faucets can mint new assets. Regular accounts can be mutable or immutable, which simply means that the account code can change after creation or not. 

Reviewer: ```suggestion
There are two basic account types in Miden. Regular accounts and faucets. Only faucets can mint new assets. Regular accounts can be mutable or immutable, which simply means that the account code can change after creation or not. 
```

Type and mutability is encoded in the most significant bits of the account's ID. 

| | Basic mutable | Basic immutable | Fungible faucet | Non-fungible faucet |
|---|---|---|---|---|
| **Description** | For most users, e.g. a wallet. Code changes allowed, including public API. | For most smart contracts. Once deployed code is immutable. | Users can issue fungible assets and customize them. | Users can issue non-fungible assets and customize them. |
| **Code updatability** | yes | no | no | no |
| **Most significant bits** | `00` | `01` | `10` | `11` |

## Public and private accounts

> Users can decide whether to keep their accounts private or public at account creation. The account ID encodes the preference on the third and fourth most significant bit.

Reviewer: ```suggestion
Users can decide whether to keep their accounts private or public at account creation. The account ID encodes this preference on the third and fourth most significant bit.
```

> * Accounts with **public state**: the actual state is stored on-chain. These would be similar to how accounts work in public blockchains, like Ethereum. Smart contracts that depend on public shared state should be stored public on Miden, e.g., DEX contract.
> * Accounts with **private state**: only the hash of the account is stored on-chain. Users who want stay private and take care of their own data should choose this mode. Users who want to interact with private accounts need to know the account's interface.

Reviewer: ```suggestion
* Accounts with public state: The actual state is stored on-chain. These would be similar to how accounts work in public blockchains, like Ethereum. Smart contracts that depend on public shared state should be stored public on Miden, e.g., DEX contract.
* Accounts with private state: Only the hash of the account is stored on-chain. Users who want stay private and take care of their own data should choose this mode. Users who want to interact with private accounts need to know the account's interface.
```

Reviewer: - Better to avoid "would"; all modals where possible in fact. 
- The word "mode" is a bit confusing for a general term.

> In the future we will also support **encrypted state** which will be on-chain but encrypted. 

Reviewer: I would avoid any statements about the future which become problematic in technical docs. What tends to happen if you allow these is docs become full of statements about the future, which often don't transpire, and it's very difficult to keep track of them and update them. Better to leave this for blog posts and marketing material.

Author: good point
